File: MySQL Database Helper/DBHelper.py
import mysql.connector
from mysql.connector import Error

class DBHelper:

    def __init__(self, user, password, database, host='localhost'):
        self.host = host
        self.user = user
        self.password = password
        self.database = database

    def getConnection(self):
        conn = None
        try:
            conn = mysql.connector.connect(
                host=self.host,
                user=self.user,
                password=self.password,
                database=self.database)

            if conn.is_connected():
                print('Connection is successfull')

        except Error as e:
            print(e)

        return conn

    def query(self, connection):
        cursor = connection.cursor()
        SQL = "SELECT ID, Name,CountryCode, District, Population FROM city"
        try:
            cursor.execute(SQL)
            results = cursor.fetchall()

            for result in results:
                print(result)

            cursor.close()
        except Error as e:
            print('Error : ' + e)


    def insert(self, connection, countryName, countryCode, district, population):
        cursor = connection.cursor()
        SQL = "INSERT INTO city (Name, CountryCode, District, Population) VALUES (%s, %s, %s, %s)"
        values = (countryName, countryCode, district, population)

        try:
            cursor.execute(SQL, values)
            connection.commit()
            cursor.close()
            print("Insert was successful")
        except Error as e:
            print(e)


    def where(self, connection, population):
        cursor = connection.cursor()
        SQL = "SELECT * FROM city WHERE Population > %s"
        val = (population, )
        cursor.execute(SQL, val)
        results = cursor.fetchall()

        for result in results:
            print(result)

        cursor.close()


    def orderby(self, connection):
        cursor = connection.cursor()
        SQL = "SELECT * FROM city ORDER BY Population"
        cursor.execute(SQL)

        results = cursor.fetchall()
        for result in results:
            print(result)

        cursor.close()


    def delete(self, connection, countryName):
        cursor = connection.cursor()
        SQL = "DELETE FROM city WHERE Name = %s"
        city = (countryName, )

        try:
            cursor.execute(SQL, city)
            connection.commit()
            print("record(s) deleted")
        except Error as e:
            print(e)
        finally:
            cursor.close()


    def update(self, connection, population, countryName):
        cursor = connection.cursor()
        SQL = "UPDATE city SET Population = %s WHERE Name = %s"
        val = (population, countryName)

        try:
            cursor.execute(SQL, val)
            connection.commit()
            print("record updated")
        except Error as e:
            print(e)
        finally:
            cursor.close()

File: MySQL Helper/DBHelper.java
import javax.swing.plaf.nimbus.State;
import java.sql.*;

public class DBHelper{

    String userName, password;
    String dbUrl = "jdbc:mysql://localhost:3306/world";

    DBHelper(String userName, String password) {
        this.userName = userName;
        this.password = password;
    }

    public Connection getConnection () throws SQLException {
        Connection conn = null;

        try {
            conn = DriverManager.getConnection(dbUrl, userName, password);
            System.out.println("Connection was successful");
        } catch (SQLException e) {
            System.out.println(e.getMessage());
        }
        return conn;
    }

    public void query(Connection conn) throws SQLException{
        Statement statement = null;
        String query = "SELECT ID, Name, CountryCode, District, Population FROM city";
        try {
            statement = conn.createStatement();
            ResultSet resultSet = statement.executeQuery(query);
            while (resultSet.next()) {
                System.out.print(resultSet.getInt("ID") + " ");
                System.out.print(resultSet.getString("Name") + " ");
                System.out.print(resultSet.getString("CountryCode") + " ");
                System.out.print(resultSet.getString("District") + " ");
                System.out.print(resultSet.getInt("Population") + " ");
                System.out.println();
            }
        } catch (SQLException e) {
            System.out.println("Error Code : " + e.getErrorCode());
            System.out.println("Error Message : " + e.getMessage());
        }
        finally {
            statement.close();
        }
    }

    public void insert(Connection connection,String name, String countryCode, String district, int population) throws SQLException {
        PreparedStatement preparedStatement = null;
        String SQL = "INSERT INTO city (Name, CountryCode, District, Population) values(?, ?, ?, ?)";
        try {
            preparedStatement = connection.prepareStatement(SQL);

            preparedStatement.setString(1, name);
            preparedStatement.setString(2, countryCode);
            preparedStatement.setString(3, district);
            preparedStatement.setInt(4, population);

            preparedStatement.executeUpdate();

            System.out.println("insert was successful");
            preparedStatement.close();
        } catch (SQLException e) {
            System.out.println("Error Code : " + e.getErrorCode());
            System.out.println("Error Message : " + e.getMessage());
        }

    }

    public void update(Connection connection, int population, String cityName) {
        PreparedStatement preparedStatement = null;
        String query;
        try {
            query = "UPDATE city set population=? WHERE Name=?";
            preparedStatement = connection.prepareStatement(query);
            preparedStatement.setInt(1, population);
            preparedStatement.setString(2, cityName);
            preparedStatement.executeUpdate();
            System.out.println("Data is updated");
        } catch (SQLException e) {
            System.out.println("Error Code : " + e.getErrorCode());
            System.out.println("Error Message : " + e.getMessage());
        }
    }

    public void delete(Connection connection, String cityName) {
        PreparedStatement preparedStatement = null;
        String SQL;
        try {
            SQL = "DELETE FROM city WHERE Name = ?";
            preparedStatement = connection.prepareStatement(SQL);
            preparedStatement.setString(1, cityName);
            preparedStatement.executeUpdate();
            System.out.println("Data is deleted");
        } catch (SQLException e) {
            System.out.println("Error Code : " + e.getErrorCode());
            System.out.println("Error Message : " + e.getMessage());
        }
    }

}
